File: Lab3/Player.java
public class Player {
    // Instance variables
    protected boolean gender;
    protected String name;
    protected int age;
    protected Country nationality;
    protected int matchesPlayed;
    protected int goals;

    // Constructor
    public Player(boolean gender, String name, int age, Country nationality) {
        this.gender = gender;
        this.name = name;
        this.age = age;
        this.nationality = nationality;
    }

    // Getters for player attributes
    public boolean getGender() {
            return gender; // Retorna gender
        }
    

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }

    public String getNationality() {
        return Country.class.getName();
    }

    // Getters for player statistics
    public int getMatchesPlayed() {
        return matchesPlayed;
    }


    public int getGoals() {
        return goals;
    }

    public void marcarGol() {
        this.goals++;
    }

    public void updateStats(Match m){
    }

    public boolean equals(Object o){
        if(o instanceof Player){
            Player c = (Player)o;
            return name == c.name;
        }
        else return false;
    }

}

File: Lab5/Player.cpp
// Player.cpp
#include "Player.hpp"

Player::Player(bool gender, std::string name, int age, Country nationality)
    : gender(gender), name(name), age(age), nationality(nationality), matchesPlayed(0), goals(0) {}

bool Player::getGender() { return gender; }

std::string Player::getName() { return name; }

int Player::getAge() { return age; }

std::string Player::getNationality() { return nationality.getName(); }

int Player::getMatchesPlayed() { return matchesPlayed; }

int Player::getGoals() { return goals; }

void Player::marcarGol() { goals++; }

bool Player::equals(Player* o) { return name == o->getName(); }
